tools/contrib/fsurdat_modifier.py

```diff
@@ -0,0 +1,356 @@
+
+
+    @staticmethod
+    def add_tag_to_filename(filename, tag):
+        """
+        Add a tag and replace timetag of a filename
+        # Expects file to end with [._]cYYMMDD.nc or [._]YYMMDD.nc
+        # Add the tag to just before that ending part
+        # and change the ending part to the current time tag
+        """
+
+        basename = os.path.basename(filename)
+        cend = -10
+
+        if ( basename[cend] == "c" ):
+           cend = cend - 1
+        if ( (basename[cend] != ".") and (basename[cend] != "_") ):
+           print ( "Trouble figuring out where to add tag to filename:" + filename )
+           os.abort()
+
+        today = date.today()
+        today_string = today.strftime("%y%m%d")
+
+        return(basename[:cend] + "_" + tag + "_c" + today_string + '.nc')
+
+
+    @staticmethod
+    def update_metadata(nc, filename):
+
+        #update attributes
+        today = date.today()
+        today_string = today.strftime("%Y-%m-%d")
+
+        #get git hash
+        sha = get_git_sha()
+
+        nc.attrs['Created_on'] = today_string
+        nc.attrs['Created_by'] = myname
+        nc.attrs['Created_with'] = os.path.abspath(__file__) + " -- " + sha
+        nc.attrs['Created_from'] = filename
+
+        #delete unrelated attributes if they exist
+        del_attrs = ['source_code', 'SVN_url', 'hostname', 'history'
+                     'History_Log', 'Logname', 'Host', 'Version',
+                     'Compiler_Optimized']
+        attr_list = nc.attrs
+
+        for attr in del_attrs:
+            if attr in attr_list:
+                del(nc.attrs[attr])
+
+
+    def modify(self):
+
+        print ("Creating surface dataset file")
+        filename = self.fsurf_in
+        print("Open file: " + filename)
+        f1 = xr.open_dataset(filename)
+
+        # modify surface data properties
+        if self.overwrite_single_pft:
+            f1['PCT_NAT_PFT'][:,:,:] = 0
+            f1['PCT_NAT_PFT'][:,:,self.dominant_pft] = 100
+        if self.zero_nonveg_landunits:
+            f1['PCT_NATVEG'][:,:]  = 100
+            f1['PCT_CROP'][:,:]    = 0
+            f1['PCT_LAKE'][:,:]    = 0.
+            f1['PCT_WETLAND'][:,:] = 0.
+            f1['PCT_URBAN'][:,:,]   = 0.
+            f1['PCT_GLACIER'][:,:] = 0.
+        if self.uniform_snowpack:
+            f1['STD_ELEV'][:,:] = 20.
+        if self.no_saturation_excess:
+            f1['FMAX'][:,:] = 0.
+
+        # specify dimension order 
+        f1= f1.transpose(u'time', u'cft', u'lsmpft', u'natpft', u'nglcec', u'nglcecp1', u'nlevsoi', u'nlevurb', u'numrad', u'numurbl', 'lsmlat', 'lsmlon')
+        
+        #update attributes
+        self.update_metadata(f1, filename)
+
+        # mode 'w' overwrites file if it exists
+        f1.to_netcdf(path=self.fsurf_out, mode='w')
+        print('Successfully created file (fsurf_out) :' + self.fsurf_out)
+        f1.close()
+
+
+def setup_logging(log_file, log_level):
```

Comment:
- [ ] This would be another good candidate for general functionality to be shared between CTSM scripts. Actually, I have already started such a module: https://github.com/escomp/CTSM/blob/master/python/ctsm/ctsm_logging.py . There is extra stuff done in this setup_logging function; we should discuss what here we really want.